improver_tests/categorical/decision_tree/test_ApplyDecisionTree.py

```diff
@@ -56,6 +57,22 @@
 from . import deterministic_diagnostic_tree, wxcode_decision_tree
 
 
+class HaltExecution(Exception):
+    pass
+
+
+@patch("improver.categorical.decision_tree.as_cubelist")
+def test_as_cubelist_called(mock_as_cubelist):
+    mock_as_cubelist.side_effect = HaltExecution
+    try:
+        ApplyDecisionTree(decision_tree=deterministic_diagnostic_tree())(
+            sentinel.precip_cube, sentinel.hail_cube
+        )
+    except HaltExecution:
+        pass
+    mock_as_cubelist.assert_called_once_with(sentinel.precip_cube, sentinel.hail_cube)
+
```

Comment:
I don't understand any of this I'm afraid. I may need some pointers on mock, and whatever sentinel is, at some point.